src/main/java/sirius/biz/analytics/charts/explorer/ChartFactory.java

```diff
@@ -0,0 +1,265 @@
+     * foundation classes).
+     *
+     * @return the name of the category to which this chart belongs
+     */
+    public abstract String getCategory();
+
+    /**
+     * Returns the CSS class which specifies the icon to show.
+     *
+     * @return the CSS class (i.e. fontawesome) to determine the icon to use
+     */
+    public abstract String getIcon();
+
+    /**
+     * Returns the label to show in the overview UI.
+     *
+     * @return the label to show for this chart. By default, this will fetch the property <tt>ClassName.label</tt>
+     */
+    public String getLabel() {
+        return NLS.get(getClass().getSimpleName() + ".label");
+    }
+
+    /**
+     * Returns the description to show in the overview UI.
+     *
+     * @return the description to show for this chart. By default, this will fetch the property <tt>ClassName.description</tt>
+     */
+    public String getDescription() {
+        return NLS.getIfExists(getClass().getSimpleName() + ".description", null).orElse("");
+    }
+
+    /**
+     * Determines the label to show above the chart.
+     *
+     * @param object the currently selected object for the chart
+     * @return the label to show above the chart. By default, this is {@link #getLabel()}.
+     */
+    protected String getChartLabel(@Nullable O object) {
+        return getLabel();
+    }
+
+    /**
+     * Determines the sub-label to show above the chart.
+     *
+     * @param object the currently selected object for the chart
+     * @return the sub-label to show. By default, this will show the name of the selected object or "" if none is
+     * selected/required.
+     */
+    protected String getChartSubLabel(@Nullable O object) {
+        return object == null ? "" : object.toString();
+    }
+
+    /**
+     * Determines the description to show next to the chart.
+     *
+     * @param object the currently selected object for the chart
+     * @return the label to show next the chart. By default, this will fetch the property <tt>ClassName.description</tt>
+     * and fall back to {@link #getDescription()}.
+     */
+    protected String getChartDescription(@Nullable O object) {
+        return NLS.getIfExists(getClass().getSimpleName() + ".chartDescription", null).orElseGet(this::getDescription);
+    }
+
+    /**
+     * Collects a list of similar / matching charts to recommend.
+     *
+     * @param referenceChartConsumer a consumer to be supplied with chart classes to suggest to the user
+     */
+
+    protected abstract void collectReferencedCharts(Consumer<Class<? extends ChartFactory<O>>> referenceChartConsumer);
+
+    /**
+     * Computes the effectively visible referenced providers.
+     *
+     * @return a list of recommended {@link ChartFactory factories} which are known to be visible to the user.
+     */
+    protected List<ChartFactory<O>> referencedFactories() {
```

Comment:
In my opinion this naming does not match the normal Java naming scheme (as in it does not start with a verb).